### DISCUSSION TITLE
Does fastlane snapshot lane support ipa files instead of mobile app source code for taking screenshots.

### DISCUSSION THREAD

**Discussion by pguna** [fastlane/fastlane#16985] [Q&A]
Fastlane Team,

Does fastlane snapshot lane support **ipa files** or **.app directory** or **unsigned xcarchive file** instead of **mobile app source code** for taking screenshots. 

 _project                    /Users/Build/remote_inputs/jobInput-13006/MobileCardServices.xcodeproj_

 Can we skip above argument and input only ipa files for snapshot lane? Please let us know if its possible in fastlane.

**Comment by pguna** [comment]
@joshdholtz Could please let us know.

**Comment by Thangaraj-M** [comment]
@joshdholtz
Could you please say whether we can pass the ipa file into the snapshot lane instead of xcodeproj file?

**Reply by NeelSaysHai** [reply]
I don't think Fastlane providing that currently. Fastlane consume project Path directly and it can't take directly .xarchive is my take. 

@joshdholtz pls keep posted if you have a solution for this query.

**Comment by rogerluan** [comment]
Here are the docs: https://docs.fastlane.tools/actions/snapshot/

tl;dr: the closest you'll get is the `test_without_building` option, which requires derived data to be provided by you.
There's no option that takes `ipa` files or `.app` directory or `unsigned xcarchive` file
